classic/form/field/GeocoderComboBox.js

```diff
@@ -214,10 +221,85 @@ Ext.define('GeoExt.form.field.GeocoderComboBox', {
         me.callParent(arguments);
 
         me.on({
-            select: this.onSelect,
+            unRestrictMapExtent: me.unRestrictExtent,
+            restrictToMapExtent: me.restrictExtent,
+            select: me.onSelect,
             focus: me.onFocus,
             scope: me
         });
+
+        if (me.restrictToMapExtent) {
+            me.restrictExtent();
+        }
+    },
+
+    /**
+     * Handle restriction to viewbox: register moveend event
+     * and update params of AJAX proxy
+     */
+    restrictExtent: function() {
+        var me = this;
+        me.map.on('moveend', me.updateExtraParams, me);
+        me.updateExtraParams();
+    },
+
+    /**
+     * Update viewbox parameter based on the current map extent
+     */
+    updateExtraParams: function() {
+        var me = this;
+        var mapSize = me.map.getSize();
+        var mv = me.map.getView();
+        var extent = mv.calculateExtent(mapSize);
+        me.addMapExtentParams(extent, mv.getProjection());
+    },
+
+    /**
+     * Update map extent params of AJAX proxy.
+     *
+     * By default, 'viewbox' and 'bounded' are updated since Nominatim is the
+     * default geocoder in this class. If no projection is passed the one of
+     * the map view is used.
+     *
+     * @param {ol.Extent} extent The extend to restrict the geocoder to
+     * @param {ol.proj.Projection} projection The projection of given extent
+     */
+    addMapExtentParams: function(extent, projection) {
+        var me = this;
+        if (!projection) {
+            projection = me.map.getView().getProjection();
+        }
+        var ll = ol.proj.transform([extent[0], extent[1]],
+            projection, 'EPSG:4326');
+        var ur = ol.proj.transform([extent[2], extent[3]],
+            projection, 'EPSG:4326');
+
+        ll = Ext.Array.map(ll, function(val) {
+            return Math.min(Math.max(val, -180), 180);
+        });
+        ur = Ext.Array.map(ur, function(val) {
+            return Math.min(Math.max(val, -180), 180);
+        });
+        var viewBoxStr = [ll.join(','), ur.join(',')].join(',');
+
+        if (me.store && me.store.getProxy()) {
+            me.store.getProxy().setExtraParam('viewbox', viewBoxStr);
+            me.store.getProxy().setExtraParam('bounded', '1');
+        }
+    },
+
+    /**
+     * Remove restriction to viewbox, in particular remove viewbox
+     * and bounded parameters from AJAX proxy and un-register
+     * moveend event from map
+     */
+    unRestrictExtent: function() {
+        var me = this;
+        if (me.store && me.store.getProxy()) {
+            me.store.getProxy().setExtraParam('viewbox', undefined);
+            me.store.getProxy().setExtraParam('bounded', undefined);
```

Comment:
Shouldn't we split here too? Unbinding in `unRestrictExtent` and calling a separate function in order to cleanup mapExtentParams introduced by `addMapExtentParams`. `viewbox` and `bounded` are Nominatim parameters and should be overwriteable as well. Maybe `removeMapExtentParams` as new function?

Reply:
Alright, see 7883bdb